### ISSUE TITLE
[1, C, H, W] channels_last contiguous tensor view as [1, C, H*W] mismatch new strides 

### ISSUE THREAD

**Issue by lijing-mt** [pytorch/pytorch#113437]
### 🐛 Describe the bug

RT, refer to the following code
```
import torch
print("======== [2,3,4,5] nchw ========")
q = torch.randn((2,3,4,5))
print(q.size(),q.stride())  # torch.Size([2, 3, 4, 5]) (60, 20, 5, 1)
z = q.view(2,3,20)
print(z.size(),z.stride())  # torch.Size([2, 3, 20]) (60, 20, 1)

print("======== [2,3,4,5] nhwc ========")
q = torch.randn((2, 3, 4, 5)).to(memory_format=torch.channels_last)
print(q.size(),q.stride())   # torch.Size([2, 3, 4, 5]) (60, 1, 15, 3)
z = q.view(2,3,20)
print(z.size(),z.stride())   # torch.Size([2, 3, 20]) (60, 1, 3)

print("======== [1,3,4,5] nchw ========")
q = torch.randn((1, 3, 4, 5))
print(q.size(),q.stride())   # torch.Size([1, 3, 4, 5]) (60, 20, 5, 1)
z = q.view(1,3,20)
print(z.size(), z.stride())  # torch.Size([1, 3, 20]) (60, 20, 1)

print("======== [1,3,4,5] nhwc ========")
q = torch.randn((1, 3, 4, 5)).to(memory_format=torch.channels_last)
print(q.size(),q.stride())   # torch.Size([1, 3, 4, 5]) (60, 1, 15, 3)
z = q.view(1,3,20)
print(z.size(), z.stride())  # torch.Size([1, 3, 20]) (3, 1, 3)
```
In ` [1,3,4,5] nhwc ` case，although stride(3, 1, 3) does not affect the accuracy，is the stride (60, 1, 3) better for undetstanding？

### Versions

[pip3] mypy-extensions==1.0.0
[pip3] numpy==1.24.4
[pip3] torch==2.0.0a0+gitc263bd4
[pip3] torchvision==0.16.0+fbb4cc5
[conda] mkl                       2023.0.0                 pypi_0    pypi
[conda] mkl-devel                 2023.0.0                 pypi_0    pypi
[conda] mkl-include               2023.0.0                 pypi_0    pypi
[conda] numpy                     1.24.4                   pypi_0    pypi
[conda] torch                     2.0.0a0+gitc263bd4          pypi_0    pypi
[conda] torchvision               0.16.0+fbb4cc5           pypi_0    pypi

**Comment by albanD** [comment]
Hi,

The stride for dimensions of size 1 can be anything (since it is never used). There is no constraint on it's value so you should not rely on it being something specific.

Closing this as expected behavior but feel free to follow up if you have other questions.
